### PR TITLE
[camera_web] Add `createCamera` implementation

### PR DESCRIPTION
**Pull request by bselwe**
Adds `createCamera` implementation of the camera platform interface.

- Added `mapFacingModeToCameraType` to `CameraSettings`.
  - Maps the [camera facing mode](https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackSettings/facingMode) to a [camera type](https://github.com/flutter/plugins/blob/master/packages/camera/camera_web/lib/src/types/camera_options.dart#L131), used when creating a camera.
- Added `mapResolutionPresetToSize` to `CameraSettings`.
  - Maps the [resolution preset](https://github.com/flutter/plugins/blob/master/packages/camera/camera_platform_interface/lib/src/types/resolution_preset.dart#L8) to a size, used when creating a camera.
  - Uses the same sizes as on Android.
- Added `createCamera` implementation.
  - Uses camera metadata initialized in `availableCameras` to extract the camera type. 
  - Creates an uninitialized camera and constrains its audio, resolution, facing mode and device id with `CameraOptions`.
  - Returns the id of the created camera.

Part of https://github.com/flutter/flutter/issues/45297.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by bselwe** [comment]
@ditman @felangel @orestesgaolin

**Review by ditman — APPROVED** [review]
LGTM! I'll revert the platform_interface change for now!

**Comment by ditman** [comment]
@googlebot I consent.

File: packages/camera/camera_web/lib/src/camera_settings.dart [hunk 1]
```diff
@@ -105,4 +106,38 @@ class CameraSettings {
...
+        return CameraType.environment;
+      case 'left':
+      case 'right':
+      default:
+        return CameraType.user;
+    }
+  }
+
+  /// Maps the given [resolutionPreset] to [Size].
+  Size mapResolutionPresetToSize(ResolutionPreset resolutionPreset) {
```
[the remark this answers is not in the document]
Comment (bselwe) [reply packages/camera/camera_web/lib/src/camera_settings.dart:127]: Hmm, I can see that photobooth records the screen orientation when the photo is being captured. Then, the photo is displayed inside the AspectRatio widget with an aspect ratio based on that orientation.

**Pull request merged**